app/hooks/use-pusher-event.ts

```diff
@@ -11,11 +12,22 @@ export function usePusherEvent<T extends unknown[] = unknown[]>(
   callback: (...args: T) => void
 ) {
   const pusher = usePusher();
+  const [channel, setChannel] = useState<Channel | null>(null);
```

Comment:
i decided to make this hook a lot smarter. now it reacts properly to changes in channelName, eventName, callback, or even the pusher instance.